woocommerce/woocommerce
# Update default visibility settings for variation options

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40198



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes.
4. By default the `Variation options` table should not shows any visible icon in each row, meaning the option is not visible in product details.
5. Edit one option by clicking the `Edit` button in the row and then check `Show in product details` checkbox from the Edit modal
6. After clicking `Update` from the modal, the visible icon should be shown in the row.
7. On hover the icon a help cursor should be shown and also a tooltip saying `Visible in product details`. 
<img width="694" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/06adde6f-e0ce-4e2e-ba92-09b3feda168e">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
